app/src/main/java/eu/quelltext/mundraub/api/API.java

```diff
@@ -143,12 +164,58 @@ public static void abortOperation(int resourceId) throws ErrorWithExplanation {
         throw new ErrorWithExplanation(resourceId);
     }
 
+    protected int setAllPlantMarkersAsync(Progressable progress) throws ErrorWithExplanation {
+        double FRACTION_DOWNLOAD = 0.05;
+        double FRACTION_PARSING = 1 - FRACTION_DOWNLOAD;
+        Set<String> urls = getUrlsForAllPlants();
+        for (String url : urls) {
+        try {
+                log.d("setAllPlantMarkersAsync", url);
+                String data = httpGet(url);
+                progress.getFraction(FRACTION_DOWNLOAD / urls.size()).setProgress(1);
+                log.d("data", data.substring(0, (data.length() > 100 ? 100 : data.length())) + " " + data.length() + " bytes");
+                Progressable fraction = progress.getFraction(FRACTION_PARSING / urls.size());
+                addMarkers(data, fraction);
+                fraction.setProgress(1);
+            } catch (JSONException e) {
+                log.printStackTrace(e);
+                return R.string.error_invalid_json_for_markers;
+            } catch (IOException e) {
+                log.printStackTrace(e);
```

Comment:
currently I am getting this error:

```
javax.net.ssl.SSLException: SSL handshake aborted: ssl=0x5ffcb580: I/O error during system call, Connection reset by peer
```

URL:
```
D/NaOvoceAPI: setAllPlantMarkersAsync: https://na-ovoce.cz/api/v1/fruit/?kind=a3c4
D/NaOvoceAPI: API GET: https://na-ovoce.cz/api/v1/fruit/?kind=a3c4
```

Reply:
According to https://stackoverflow.com/questions/20741405/javax-net-ssl-sslexception-ssl-handshake-aborted-connection-reset-by-peer-while
TLS1.0 is disabled and causing this error
https://www.ssllabs.com/ssltest/analyze.html?d=na-ovoce.cz shows it is disabled

Reply:
Seems like the server needs to enable tls1.0 to allow such connections.
https://stackoverflow.com/a/44084570/1320237

Reply:
Also ssl is disabled. And there is an https redirect. @jsmesami, I cannot connect to the service. I think, because my android devices are too old. I can try to make an http proxy.
Also, using the insecure mode, this is not possible because SSL is disabled, see https://www.ssllabs.com/ssltest/analyze.html?d=na-ovoce.cz.

Reply:
What do you think?